### PR TITLE
Network Switching

### PR DESCRIPTION
**Pull request by kalverra**
Makes use of the `SelectedNetwork` in order to choose what network to run your tests on. Previously using an `Entry` for each network led to bloated code, clumsy selection, and minor performance issues. This utilizes some new environment variables, namely `SELECTED_NETWORK` that selects which supported network to run tests on. This also enables setting network-specific secrets (e.g. `GOERLI_URLS` and `GOERLI_KEYS`).

```sh
########## Network Settings ##########

# Select a pre-defined network 
export SELECTED_NETWORK="SIMULATED"

# General private values that will be retrieved when running on non-simulated networks
export EVM_URLS="wss://evm.url,wss://other.url" # Comma-sparated list of websocket URLs to use when running on live networks
export EVM_PRIVATE_KEYS="private,funding,keys" # Comma-separated list of private keys to use when running on live networks

# Specific private values retrieved when running on specified chains
# Goerli
export GOERLI_URLS="wss://goerli.io/ws"
export GOERLI_KEYS="goerli,funding,keys"
# Sepolia
export SEPOLIA_URLS="wss://sepolia.io/ws"
export SEPOLIA_KEYS="sepolia,funding,keys"
# Klaytn Baobab
export KLAYTN_BAOBAB_URLS="wss://klaytn.io/ws"
export KLAYTN_BAOBAB_KEYS="klaytn,funding,keys"
# Metis Stardust
export METIS_STARDUST_URLS="wss://metis.io/ws"
export METIS_STARDUST_KEYS="metis,funding,keys"
# Arbitrum Goerli
export ARBITRUM_GOERLI_URLS="wss://arbitrum.io/ws"
export ARBITRUM_GOERLI_KEYS="arbitrum,funding,keys"
# Optimism Goerli
export OPTIMISM_GOERLI_URLS="wss://optimism.io/ws"
export OPTIMISM_GOERLI_KEYS="optimism,funding,keys"
```

### CODE REVIEW

File: integration-tests/known_networks.go [hunk 1]
```diff
@@ -1,103 +1,162 @@
...
+		"KLAYTN_BAOBAB":   KlaytnBaobab,
+		"METIS_STARDUST":  MetisStardust,
+		"ARBITRUM_GOERLI": ArbitrumGoerli,
+		"OPTIMISM_GOERLI": OptimismGoerli,
+	}
 )
+
+// DetermineNetwork determines which network
+func determineSelectedNetwork() *blockchain.EVMNetwork {
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
```
Comment (skudasov) [inline integration-tests/known_networks.go:112]: That doesn't belong here, we had a method previously to select `LOG_LEVEL` ?

File: integration-tests/known_networks.go [hunk 2]
```diff
@@ -1,103 +1,162 @@
...
+	}
+	log.Fatal().
+		Str("SELECTED_NETWORK", setNetwork).
+		Str("Valid Networks", strings.Join(validNetworks, ", ")).
+		Msg("SELECTED_NETWORK value of is invalid. Use a listed valid one")
+	return nil
+}
+
+func getURLs(prefix string) []string {
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
```
Comment (skudasov) [inline integration-tests/known_networks.go:135]: The same in `determineSelectedNetwork`

File: integration-tests/known_networks.go [hunk 3]
```diff
@@ -1,103 +1,162 @@
...
+			Msg(fmt.Sprintf("No '%s' env var defined, defaulting to 'EVM_URLS'", envVar))
+		return urls
+	}
+	urls := strings.Split(os.Getenv(envVar), ",")
+	log.Info().Interface(envVar, urls).Msg("Read network URLs")
+	return urls
+}
+
+func getKeys(prefix string) []string {
+	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
```
Comment (skudasov) [inline integration-tests/known_networks.go:151]: The same in `determineSelectedNetwork`

File: integration-tests/example.env [hunk 4]
```diff
@@ -15,11 +15,34 @@ export SLACK_USER="U000000000" # User ID of the person running the soak tests to
 
 ########## Network Settings ##########
 
-# Private values that will be retrieved when running on non-simulated networks
-export EVM_PRIVATE_KEYS="private,funding,keys" # Comma-separated list of private keys to use when running on live networks
+# Select a pre-defined network 
+export SELECTED_NETWORK="Goerli"
```
Comment (AnieeG) [inline integration-tests/example.env:19]: If the test needs to handle more than one network, would it make sense to make it a list of networks?

Comment (kalverra) [reply integration-tests/example.env:19]: Use case being something like "I want to run OCR on Goerli and Optimism"?

Comment (AnieeG) [reply integration-tests/example.env:19]: Something like CCIP. Or probably the case where we make use of EVM_NODEs instead of EVM_CHAIN_ID

**Review by skudasov — CHANGES_REQUESTED** [review]

File: integration-tests/smoke/keeper_test.go [hunk 5]
```diff
@@ -97,7 +97,7 @@ var highBCPTRegistryConfig = contracts.KeeperRegistrySettings{
 	FallbackLinkPrice:    big.NewInt(2e18),
 }
 
-var _ = Describe("Keeper Suite @keeper", func() {
+var _ = PDescribe("Keeper Suite @keeper", func() {
```
Comment (tateexon) [inline integration-tests/smoke/keeper_test.go:100]: Did you mean to leave this as suite as pending?

Comment (kalverra) [reply integration-tests/smoke/keeper_test.go]: Fixed

**Comment by kalverra** [comment]
Now uses `SELECTED_NETWORKS`, enabling choosing multiple networks, like for use in CCIP tests.

File: integration-tests/smoke/forwarder_ocr_test.go [hunk 6]
```diff
@@ -117,7 +82,7 @@ var _ = Describe("OCR forwarder flow - each operator forwarder pair belongs to e
 		Expect(err).ShouldNot(HaveOccurred(), "Deploying Link Token Contract shouldn't fail")
 
 		By("Funding Chainlink nodes")
-		err = actions.FundChainlinkNodes(chainlinkNodes, chainClient, funding)
+		err = actions.FundChainlinkNodes(chainlinkNodes, chainClient, big.NewFloat(.05))
```
Comment (tateexon) [inline integration-tests/smoke/forwarder_ocr_test.go:85]: We had different funding values depending on the network before. Is this magic number good for all networks now some how? I see similar things in the other test files.

File: integration-tests/known_networks.go [hunk 7]
```diff
@@ -1,103 +1,184 @@
...
+		"ARBITRUM_GOERLI": ArbitrumGoerli,
+		"OPTIMISM_GOERLI": OptimismGoerli,
+	}
 )
+
+// determineSelectedNetworks uses `SELECTED_NETWORKS` to determine which network(s) to run the tests on
+func determineSelectedNetworks() []*blockchain.EVMNetwork {
+	logging.Init()
+	selectedNetworks := make([]*blockchain.EVMNetwork, 0)
+	setNetworkNames := strings.Split(strings.ToUpper(os.Getenv("SELECTED_NETWORKS")), ",")
```
Comment (tateexon) [inline integration-tests/known_networks.go:118]: I thought envconfig could automatically create these slice's for us? Should be able to do the same for urls and keys which might simplify some of this logic.

Comment (kalverra) [reply integration-tests/known_networks.go:118]: Ya, can move this to the chainlink-env logic.

File: integration-tests/smoke/forwarder_ocr_test.go (hunk 6 above)
Comment (kalverra) [reply integration-tests/smoke/forwarder_ocr_test.go:85]: So, this is tricky either way. Best to discuss it elsewhere.

**Review by AnieeG — APPROVED** [review]

**Review by skudasov — APPROVED** [review]

**Review by tateexon — APPROVED** [review]

**Review by anirudhwarrier — APPROVED** [review]

**Pull request merged**